pkg/operator/starter.go

```diff
@@ -209,3 +223,40 @@ func extractOperatorStatus(obj *unstructured.Unstructured, fieldManager string)
 	}
 	return &ret.Status.OperatorStatusApplyConfiguration, nil
 }
+
+func withCustomEndPoint(infraLister v1.InfrastructureLister) dc.DeploymentHookFunc {
+	return func(_ *opv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		infra, err := infraLister.Get(infrastructureName)
+		if err != nil {
+			return err
+		}
+		if infra.Status.PlatformStatus == nil || infra.Status.PlatformStatus.PowerVS == nil {
+			return nil
+		}
+		serviceEndPoints := infra.Status.PlatformStatus.PowerVS.ServiceEndpoints
+		if len(serviceEndPoints) == 0 {
+			return nil
+		}
+		var containerEnvVars []corev1.EnvVar
+		for _, serviceEndPoint := range serviceEndPoints {
+			if _, ok := endPointKeyToEnvNameMap[serviceEndPoint.Name]; !ok {
+				klog.Infof("Ignoring %q serviceEndpoint", serviceEndPoint.Name)
+				continue
+			}
+			containerEnvVars = append(containerEnvVars, corev1.EnvVar{
+				Name:  endPointKeyToEnvNameMap[serviceEndPoint.Name],
+				Value: serviceEndPoint.URL,
+			})
+		}
+
+		for i := range deployment.Spec.Template.Spec.Containers {
+			container := &deployment.Spec.Template.Spec.Containers[i]
+			if container.Name != csiDriver {
+				continue
+			}
+			container.Env = append(container.Env, containerEnvVars...)
+			return nil
+		}
+		return nil
+	}
+}
```

Comment:
@kishen-v I tested your changes for `deployment/ibm-powervs-block-csi-driver-controller`, but found that we would need to do the same for `daemonset/ibm-powervs-block-csi-driver-node` as well. So, I tried this (very similarly named) func that deals with `DaemonSet` and called from line 174 (inside `.WithCSIDriverNodeService` block);
```
+func withCustomEndpoint(infraLister v1.InfrastructureLister) csidrivernodeservicecontroller.DaemonSetHookFunc {
+       return func(_ *opv1.OperatorSpec, daemonset *appsv1.DaemonSet) error {
+               infra, err := infraLister.Get(infrastructureName)
+               if err != nil {
+                       return err
+               }
+               if infra.Status.PlatformStatus == nil || infra.Status.PlatformStatus.PowerVS == nil {
+                       return nil
+               }
+               serviceEndPoints := infra.Status.PlatformStatus.PowerVS.ServiceEndpoints
+               if len(serviceEndPoints) == 0 {
+                       return nil
+               }
+               var containerEnvVars []corev1.EnvVar
+               for _, serviceEndPoint := range serviceEndPoints {
+                       if _, ok := endPointKeyToEnvNameMap[serviceEndPoint.Name]; !ok {
+                               klog.Infof("Ignoring %q serviceEndpoint", serviceEndPoint.Name)
+                               continue
+                       }
+                       containerEnvVars = append(containerEnvVars, corev1.EnvVar{
+                               Name:  endPointKeyToEnvNameMap[serviceEndPoint.Name],
+                               Value: serviceEndPoint.URL,
+                       })
+               }
+
+               for i := range daemonset.Spec.Template.Spec.Containers {
+                       container := &daemonset.Spec.Template.Spec.Containers[i]
+                       if container.Name != csiDriver {
+                               continue
+                       }
+                       container.Env = append(container.Env, containerEnvVars...)
+                       return nil
+               }
+               return nil
+       }
+}
```
Does it make sense? If so, please incorporate this into your PR. After running this operator locally and scaling back up `cluster-storage-operator`, the `storage` CO becomes `Available`.

Reply:
Thanks @miyamotoh. I have made the changes as suggested.